### PR TITLE
handle loss and name marching wrappers

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: Describe your changes.
ORTTrainer fails to handle cases where both WrapModel and model_loss_cls are needed.
**Motivation and Context**
fix the failing case

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_transformers.py [hunk 1]
```diff
@@ -199,9 +196,116 @@ def get_lr_this_step(global_step):
     def setUp(self):
         self.model_tester = BertModelTest.BertModelTester(self)
 
-    def test_for_pretraining(self):
+    def test_for_pretraining_mixed_precision_all(self):
+        # It would be better to test both with/without mixed precision and allreduce_post_accumulation.
+        # However, stress test of all the 4 cases is not stable at lease on the test machine.
```
Comment (BowenBao) [inline orttraining/orttraining/test/python/orttraining_test_transformers.py:201]: nit: *least

btw, why is it not stable?

Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_test_transformers.py]: if I multiple tests are run in one process, it sometimes fails with a segfault. I still need to investigate it. It is at the backend.

File: onnxruntime/test/python/onnxruntime_test_ort_trainer.py [hunk 2]
```diff
@@ -652,5 +652,37 @@ def testBertCheckpointingLoadZero(self):
...
+        loss_desc = IODescription('loss', [], torch.float32)
+        model_desc = ModelDescription([input_desc, label_desc], [loss_desc, output_desc])
+        def loss_fn(x, label):
+            return F.nll_loss(F.log_softmax(x, dim=1), label)
+        ort_trainer = ORTTrainer(
+            pt_model, loss_fn, model_desc, "SGDOptimizer", None,
+            IODescription('Learning_Rate', [1, ], torch.float32), device,
+            _opset_version=10)
+        learningRate = 0.02
+        ort_trainer.train_step(x=data, label=label, Learning_Rate=torch.tensor([learningRate]))
```
Comment (BowenBao) [inline onnxruntime/test/python/onnxruntime_test_ort_trainer.py:683]: nit: should we also test internal lr as well?

File: onnxruntime/test/python/onnxruntime_test_ort_trainer.py [hunk 3]
```diff
@@ -704,5 +704,43 @@ def testBertCheckpointingLoadZero(self):
...
+        def get_lr_this_step(global_step):
+            learningRate = 0.02
+            return torch.tensor([learningRate])
+
+        ort_trainer = ORTTrainer(
+            pt_model, loss_fn, model_desc, "SGDOptimizer", None,
+            IODescription('Learning_Rate', [1, ], torch.float32), device,
+            get_lr_this_step=get_lr_this_step,
+            _opset_version=10)
+        learningRate = 0.02
```
Comment (BowenBao) [inline onnxruntime/test/python/onnxruntime_test_ort_trainer.py:740]: nit: unused

File: onnxruntime/test/python/onnxruntime_test_ort_trainer.py [hunk 4]
```diff
@@ -704,5 +704,43 @@ def testBertCheckpointingLoadZero(self):
...
+        
+        def get_lr_this_step(global_step):
+            learningRate = 0.02
+            return torch.tensor([learningRate])
+
+        ort_trainer = ORTTrainer(
+            pt_model, loss_fn, model_desc, "SGDOptimizer", None,
+            IODescription('Learning_Rate', [1, ], torch.float32), device,
+            get_lr_this_step=get_lr_this_step,
+            _opset_version=10)
```
Comment (BowenBao) [inline onnxruntime/test/python/onnxruntime_test_ort_trainer.py:739]: nit: should we test 12? (or leave as default)
